### PR TITLE
ActionButtons Feature - Stage 1. Collect all existing action buttons in one place

### PR DESCRIPTION
**Pull request by churkin**

### CODE REVIEW

File: js/ui/drop_down_editor/ui.drop_down_button.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        const $element = $("<div>").addClass(DROP_DOWN_EDITOR_BUTTON_CLASS);
+
+        const instance = editor._createComponent($element, Button, {
+            focusStateEnabled: false,
+            hoverStateEnabled: false,
+            activeStateEnabled: false,
+            useInkRipple: false,
+        });
+
+        $element.removeClass("dx-button");
```
Comment (DokaRus) [inline js/ui/drop_down_editor/ui.drop_down_button.js]: Can we save `dx-button` class? Perhaps we can use the `stylingMode: "text"` [option](https://js.devexpress.com/Demos/WidgetsGallery/Demo/Button/PredefinedTypes/jQuery/Light/) to get a desired appearance?

File: js/ui/drop_down_editor/ui.drop_down_button.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+        return {
+            $element,
+            instance
+        };
+    }
+
+    _isVisible() {
```
Comment (DokaRus) [inline js/ui/drop_down_editor/ui.drop_down_button.js]:
```suggestion
    get _isVisible() {
```

File: js/ui/drop_down_editor/ui.drop_down_button.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+        const { editor, instance } = this;
+        const isRendered = !!instance;
+
+        if(isRendered) {
+            const $editor = editor.$element();
+            const isVisible = this._isVisible();
+            const isReadOnly = editor.option("readOnly");
+
+            instance.option("visible", isVisible);
```
Comment (DokaRus) [inline js/ui/drop_down_editor/ui.drop_down_button.js]:
```suggestion
            instance.option({ 
                visible: isVisible,
                disabled: isReadOnly,
                template: editor._getTemplateByOption("dropDownButtonTemplate")
            });
```

File: js/ui/number_box/number_box.spins.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        if(isRendered) {
+            const $editor = editor.$element();
+            const isVisible = this._isVisible();
+            const isDisabled = editor.option("disabled");
+            const isTouchFriendly = editor.option("showSpinButtons") && editor.option("useLargeSpinButtons");
+            const $spinButtons = instance.children();
+            const spinUp = SpinButton.getInstance($spinButtons.eq(0));
+            const spinDown = SpinButton.getInstance($spinButtons.eq(1));
+
+            spinUp.option("disabled", isDisabled);
```
Comment (DokaRus) [inline js/ui/number_box/number_box.spins.js]:
```suggestion
            spinUp.option({
                disabled: isDisabled,
                visible: isVisible
            });
            spinDown.option({
                disabled: isDisabled,
                visible: isVisible
            });
            $editor
                .toggleClass(SPIN_TOUCH_FRIENDLY_CLASS, isTouchFriendly)
                .toggleClass(SPIN_CLASS, isVisible);
```

File: js/ui/text_box/action_button_collection/button.base.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    _shouldRender() {
+        return this._isVisible() && !this._isRendered();
+    }
+
+    dispose() {
+        const { instance, $placeMarker } = this;
+
+        if(instance) {
+            // TODO: instance.dispose()
+            instance.dispose ? instance.dispose() : instance.remove();
```
Comment (DokaRus) [inline js/ui/text_box/action_button_collection/button.base.js]: >instance.remove()

Is it related to the clear button?

File: js/ui/text_box/ui.text_editor.clear.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        super("clear", editor, options);
+    }
+
+    _createInstance() {
+        const $element = $("<span>")
+            .addClass(TEXTEDITOR_CLEAR_BUTTON_CLASS)
+            .append($("<span>").addClass(TEXTEDITOR_ICON_CLASS).addClass(TEXTEDITOR_CLEAR_ICON_CLASS));
+
+        return {
+            instance: $element,
```
Comment (DokaRus) [inline js/ui/text_box/ui.text_editor.clear.js:28]: Can we provide an object with the interface like a widget instance?
```suggestion
            instance: {
                element: () => $element,
                dispose: () => $element.remove()
            },
```

File: js/ui/text_box/action_button_collection/index.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        this.buttons = [];
+        this.defaultButtonsInfo = defaultButtonsInfo;
+        this.editor = editor;
+    }
+
+    _createButton(buttonName) {
+        const defaultButtonInfo = this.defaultButtonsInfo.find(({ name }) => name === buttonName);
+
+        if(!defaultButtonInfo) {
+            throw "Can't create custom action button";
```
Comment (DokaRus) [inline js/ui/text_box/action_button_collection/index.js:16]: ToDo: use `ui/widget/ui.errors` to generate an error message

**Review by mrjohn0011 — APPROVED** [review]

**Review by DokaRus — APPROVED** [review]
ToDo: add\extract separate tests for the `action_button_collection` and `action_button_collection\button`

**Pull request merged**